### PR TITLE
[CI] Enable wolfi Elastic Agent images in tests

### PR DESCRIPTION
**Pull request by mrodm**
Run tests using Elastic Agent images where it is possible.

## Author's checklist
- [x] Remove update DefaultVersion variable
- [x] Check integration tests with 8.16.0-SNAPSHOT
    - https://buildkite.com/elastic/elastic-package/builds/3914
    - https://github.com/elastic/elastic-package/pull/2067/commits/8b742988e55e0a8431f440f74ddea79fb2d17e72

### CODE REVIEW

File: internal/install/stack_version.go [hunk 1]
```diff
@@ -6,5 +6,5 @@ package install
 
 const (
 	// DefaultStackVersion is the default version of the stack
-	DefaultStackVersion = "8.15.0"
+	DefaultStackVersion = "8.16.0-SNAPSHOT"
```
Comment (mrodm) [inline internal/install/stack_version.go:9]: To be removed

Comment (mrodm) [inline internal/install/stack_version.go:9]: To be removed

File: test/packages/parallel/apache/data_stream/access/_dev/test/policy/test-default.expected [hunk 2]
```diff
@@ -17,6 +17,7 @@ inputs:
             - apache-access
       type: logfile
       use_output: default
+namespaces: []
```
Comment (mrodm) [inline test/packages/parallel/apache/data_stream/access/_dev/test/policy/test-default.expected:20]: It looks like that for policy tests it should also be needed to be able to mark some fields optional?
This `namespaces` field appeared when testing with `8.16.0-SNAPSHOT`.

Changes required present in these two commits:
- https://github.com/elastic/elastic-package/pull/2067/commits/ba7e1f5f05f4cae7ac91835a94800982dc62490b
- https://github.com/elastic/elastic-package/pull/2067/commits/83a8623bf19143627eec25bdf3b5c5f9bdaf6dfc

Comment (mrodm) [reply test/packages/parallel/apache/data_stream/access/_dev/test/policy/test-default.expected:20]: Reverting these two commits in order to be able to merge this PR.

File: internal/install/application_configuration_test.go [hunk 3]
```diff
@@ -52,8 +53,12 @@ func TestSelectElasticAgentImageName_NextStackInOwnNamespace(t *testing.T) {
 	assert.Equal(t, selected, elasticAgentCompleteImageName)
 }
 
-func TestSelectElasticAgentImageName_DefaultImage816(t *testing.T) {
+func TestSelectElasticAgentImageName_DefaultImage816_WihtoutEnvVar(t *testing.T) {
 	version := stackVersion8160
+	// Try to keep the test agnostic from the environment variables defined in CI
+	t.Setenv(disableElasticAgentWolfiEnvVar, "")
+	os.Unsetenv(disableElasticAgentWolfiEnvVar)
```
Comment (mrodm) [inline internal/install/application_configuration_test.go:60]: Could this cause any issues ?
It looks like the environment variable is defined in other tests even here it is unset.

Comment (jsoriano) [reply internal/install/application_configuration_test.go:60]: Setting environment variables is only going to be problematic if tests are run in parallel, as it affects the global process. `t.Setenv` recovers the previous value on cleanup, see https://pkg.go.dev/testing#B.Setenv.

Comment (mrodm) [reply internal/install/application_configuration_test.go:60]: Perfect, so as these tests do not use `t.Parallel()`, I guess it is safe here. Even if it is called `os.Unsetenv` , the environment variable would be recovered aftwards thanks to the t.Setenv()

Comment (jsoriano) [reply internal/install/application_configuration_test.go:60]: I think each package is executed in a different process, so we should be fine if the tests are in different packages.

**Review by jsoriano — APPROVED** [review]

**Pull request merged**